collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -352,6 +287,68 @@ private void getFiles() {
         getFileTask.execute(toDownload);
     }
 
+    @Override
+    protected void updateAdapter() {
+        CharSequence charSequence = getFilterText();
+        filteredList.clear();
+
+        if (charSequence.length() > 0) {
+            for (DriveListItem item : driveList) {
+                if (item.getName().toLowerCase(Locale.US).contains(charSequence.toString().toLowerCase(Locale.US))) {
+                    filteredList.add(item);
+                }
+            }
+        } else {
+            filteredList.addAll(driveList);
+        }
+
+        sortList();
+        if (adapter == null) {
+            adapter = new FileArrayAdapter(this, filteredList);
+            listView.setAdapter(adapter);
+        } else {
+            adapter.notifyDataSetChanged();
+        }
+
+        adapter.notifyDataSetChanged();
+        checkPreviouslyCheckedItems();
+    }
+
+    @Override
+    protected void checkPreviouslyCheckedItems() {
+        listView.clearChoices();
+        for (int i = 0; i < listView.getCount(); i++) {
+            DriveListItem item = (DriveListItem) listView.getAdapter().getItem(i);
+            if (toDownload.contains(item)) {
+                listView.setItemChecked(i, true);
+            }
+        }
+    }
+
+    private void sortList() {
+        Collections.sort(filteredList, new Comparator<DriveListItem>() {
+            @Override
+            public int compare(DriveListItem lhs, DriveListItem rhs) {
+                if (lhs.getType() != rhs.getType()) {
+                    if (lhs.getType() == DriveListItem.DIR) {
+                        return -1;
+                    } else {
+                        return 1;
+                    }
+                } else if (getSortingOrder().equals(SORT_BY_NAME_ASC)) {
```

Comment:
Above 5 lines can be written in a single line
`return lhs.getType() == DriveListItem.DIR ? -1 : 1;`

Reply:
Or maybe something like this:

```
if (lhs.getType() != rhs.getType()) {
                    return lhs.getType() == DriveListItem.DIR ? -1 : 1;
                } else {
                    int compareName = lhs.getName().compareToIgnoreCase(rhs.getName());
                    return getSortingOrder().equals(SORT_BY_NAME_ASC) ? compareName : -compareName;
                }
```